server/auth/auth.go

```diff
@@ -190,6 +196,104 @@ type AuthManager struct {
 
 	pendingMessagesMtx sync.Mutex
 	pendingMessages    map[account.AccountID]map[uint64]*timedMessage
+
+	violationMtx    sync.Mutex
+	violationCounts map[account.AccountID]violationCounter
+}
+
+// violation badness
+const (
+	// preimage miss
+	preimageMissScore = 2 // book spoof, no match, no stuck funds
+
+	// failure to act violations
+	noSwapAsMakerScore   = 4  // book spoof, match with taker order affected, no stuck funds
+	noSwapAsTakerScore   = 11 // maker has contract stuck for 20 hrs
+	noRedeemAsMakerScore = 7  // taker has contract stuck for 8 hrs
+	noRedeemAsTakerScore = 1  // just dumb, counterparty not inconvenienced
+
+	successScore = -1 // offsets the violations
+	successCap   = 30
+
+	defaultBanScore = 20
```

Comment:
"What's the minus of 30 and 20?"

So if someone racks up 30 successes, they have a score of -30 to work with, meaning 50 points until they get banned.  By comparison, a new user would have just 20.  Such a jacked up user could significantly inconvenience 5 makers (landing at 55) or 8 takers (landing at 56) or combinations before getting banned.  Maybe a lower successCap?

Reply:
If the success cap is defined as **the most historical success points that can be applied to the score**, then older/more active users would be unfairly disadvantaged, because a user can rack up infinite successes and 50 ban points and be banned the same as someone with 30 successes and 50 ban points.

Reply:
Right, but I thought that addressed was the essence of the issue you raised in https://github.com/decred/dcrdex/pull/700#pullrequestreview-497708754.  Can you clarify what you meant by "Violations and successes should only be counted back some maximum depth"?

I figure if a well established user suddenly starts failing a lot of swaps, we'd want to handle that in the same way as a less well established user that suddenly starts failing a lot of swaps, with the resolution for both cases being operator intervention to reinstate the account after the reason for the violations has been established *and* resolved so that it does not continue to happen (e.g. their setup somehow went sideways).  If either user has no legit explanation for their actions, the outcome would be the same.

Reply:
> we'd want to handle that in the same way as a less well established user that suddenly starts failing a lot of swaps

But you're also catching the case where an active or long existing user fails a swap only every once in a while. The successes are dropping off but the failures can accumulate forever. If you want to catch only transient issues, you'll need to limit the window. By "some maximum depth" I mean some maximum count of orders or matches, like we do with the cancellation statistics.

Reply:
That's clear now.  I'll revert successCap and toy with some ideas for implementing maximum depth as you've described it.

Reply:
The maximum count/depth of orders or matches was pretty brutal to implement, but it's in https://github.com/decred/dcrdex/pull/700/commits/8e307d560b4a7f999937d12b75e3dcd44e0466f9.  Gonna take a little more time for `pgonline` tests and/or interactive tests, but unit tests past and the queries were running as expected manually in psql.